PurchasesCoreSwift/FoundationExtensions/DateFormatter+Extensions.swift

```diff
@@ -0,0 +1,35 @@
+//
+// Created by Andrés Boedo on 7/29/20.
+// Copyright (c) 2020 Purchases. All rights reserved.
+//
+
+import Foundation
+
+extension DateFormatter {
+    static let iso8601SecondsDateFormatter: DateFormatter = {
+        let dateFormatter = DateFormatter()
+        dateFormatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ssZ"
+        return dateFormatter
+    }()
+
+    static func iso8601SecondsOrMillisecondsDate(fromBytes bytes: ArraySlice<UInt8>) -> Date? {
+        guard let dateString = String(bytes: Array(bytes), encoding: .ascii) else { return nil }
+        return (Self.iso8601SecondsDateFormatter.date(from: dateString)
+            ?? Self.iso8601MilliSecondsDateFormatter.date(from: dateString))
+    }
+
+    func date(fromString maybeDateString: String?) -> Date? {
+        guard let dateString = maybeDateString else { return nil }
```

Comment:
any reason this func can't be static too?

...that might've been me, but only just noticing

Reply:
I figured that since this `func` makes no assumptions regarding the format of the date, it's better to keep it as an instance method - this way it would also work for non-iso8601 dates